### ISSUE TITLE
Query lang problems

### ISSUE THREAD

**Issue by gtfierro** [gtfierro/giles#19]
`select distinct Metadata/HVACZone where Metadata/System = 'HVAC';` gives following stacktrace

```
INFO Feb 15 18:46:35 archiver.go:179 ▶ select distinct Metadata/HVACZone where Metadata/System = 'HVAC';
2015/02/15 18:46:35 http: panic serving 127.0.0.1:48173: runtime error: index out of range
goroutine 4094 [running]:
net/http.func·011()
        /usr/local/go/src/net/http/server.go:1130 +0xbb
github.com/gtfierro/giles/archiver.getnodeAt(0x3, 0xc20847ee40, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1)
        /home/gabe/go/src/github.com/gtfierro/giles/archiver/ast.go:263 +0x474
github.com/gtfierro/giles/archiver.parseWhere(0xc20847ee40, 0x8355a0)
        /home/gabe/go/src/github.com/gtfierro/giles/archiver/ast.go:240 +0x6a7
github.com/gtfierro/giles/archiver.makeAST(0xc2081ac080, 0x7, 0x8, 0x7, 0x0, 0x0)
        /home/gabe/go/src/github.com/gtfierro/giles/archiver/ast.go:307 +0x2ad
github.com/gtfierro/giles/archiver.parse(0xc2081160f0, 0x41, 0x41)
        /home/gabe/go/src/github.com/gtfierro/giles/archiver/ast.go:314 +0x63
github.com/gtfierro/giles/archiver.(*Archiver).HandleQuery(0xc2080d19a0, 0xc2081160f0, 0x41, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/gabe/go/src/github.com/gtfierro/giles/archiver/archiver.go:181 +0x1f4
github.com/gtfierro/giles/httphandler.QueryHandler(0xc2080d19a0, 0x7f89c1b7c160, 0xc2081563c0, 0xc2080b2000, 0x0, 0x0, 0x0)
        /home/gabe/go/src/github.com/gtfierro/giles/httphandler/handlers.go:119 +0x356
github.com/gtfierro/giles/httphandler.func·001(0x7f89c1b7c160, 0xc2081563c0, 0xc2080b2000, 0x0, 0x0, 0x0)
        /home/gabe/go/src/github.com/gtfierro/giles/httphandler/handlers.go:46 +0x6d
github.com/julienschmidt/httprouter.(*Router).ServeHTTP(0xc2081082d0, 0x7f89c1b7c160, 0xc2081563c0, 0xc2080b2000)
        /home/gabe/go/src/github.com/julienschmidt/httprouter/router.go:298 +0x18e
net/http.(*ServeMux).ServeHTTP(0xc20803cae0, 0x7f89c1b7c160, 0xc2081563c0, 0xc2080b2000)
        /usr/local/go/src/net/http/server.go:1541 +0x17d
net/http.serverHandler.ServeHTTP(0xc2080aa000, 0x7f89c1b7c160, 0xc2081563c0, 0xc2080b2000)
        /usr/local/go/src/net/http/server.go:1703 +0x19a
net/http.(*conn).serve(0xc2081ba1e0)
        /usr/local/go/src/net/http/server.go:1204 +0xb57
created by net/http.(*Server).Serve
        /usr/local/go/src/net/http/server.go:1751 +0x35e

```

**Comment by gtfierro** [comment]
Looks like this is caused when the "distinct" keyword is used -- the parseWhere function includes the "where" string, and this offsets everything.

**Comment by gtfierro** [comment]
Should be "fixed" by 6d71ad577b5c0606bee5e17fdf2d01212d02170b
